client/src/app/App.js

```diff
@@ -8,6 +8,8 @@ import {
   reduce
 } from 'min-dash';
 
+import executeOnceWithTab from './util/executeOnceWithTab';
```

Comment:
If you make this a global utility, please get rid of the tab in it:

```js
function executeOnce(fn, identity) {

   return function(...args) {
     const id = identity(...args);

     if (...) { }
   }
}
```

Using it for the tabs specifically:

```
this.checkStuff = executeOnce(this.checkStuff, (tab) => tab.id);
```